File: frontend/src/types/product.ts
import type { BaseResponse } from "./common"

interface ProductSeries extends BaseResponse {
  id: number
  name: string
}

interface Product extends BaseResponse {
  id: number
  name: string
  seriesId: string
  image: string
  description: string
  cover: string
  previews: string[]
}

interface ProductDetail extends BaseResponse {
  id: number
  name: string
  description: string
  cover: string
  previews: string[] | null

  series_id: number
  series_name: string
}

export type { ProductSeries, Product, ProductDetail }

File: backend/pkg/response/product.go
package response

import (
	"time"

	"github.com/Smash249/xenova/backend/internal/models"
)

type ProductDetailResp struct {
	ID          uint              `json:"id"`
	SeriesID    uint              `json:"series_id"`
	SeriesName  string            `json:"series_name"`
	Name        string            `json:"name"`
	Description string            `json:"description"`
	Cover       string            `json:"cover"`
	Previews    models.StringList `json:"previews"`
	Price       float64           `json:"price"`
	CreatedAt   time.Time         `json:"created_at"`
	UpdatedAt   time.Time         `json:"updated_at"`
}
